### ISSUE TITLE
http tunneling using GET method instead websocket

### ISSUE THREAD

**Issue by ingnelson** [jpillora/chisel#174]
Hello master.

what is the option to use GET method to create the tunnel ?

I firewall blocks websocket protocol , only accept HTTP , TCP or HTTP over TCP .

So i need the first a HTTP request and the server can send the first response via HTTP then everything using TCP protocol.

I will wait your answer , because i really need it.

**Comment by ironiridis** [comment]
How do you know your firewall doesn't block TCP traffic after the first request? Do you have code demonstrating what works for your firewall?

**Comment by ingnelson** [comment]
@ironiridis i know my firewall do not block TCP because i did many test using different method and different protocols .
I found that my ISP firewall only let pass HTTP and TCP protocol with this HTTP method on the request : 
`GET , HEAD , DELETE , POST  `

if i use `CONNECT` method the ISP firewall blocks the HTTP request and the server or the the VPS do not receive the HTTP request and the connection is blocked.

I did test with `echo`  and `curl` commands using the above HTTP method and i get a connection between my Android device and my VPS , this mean only the CONNECT method and the TLS protocol is blocked by my ISP provider.

Test #1: 
`(echo -en 'GET #abc# HTTP/1.1\r\n ٌُُُُُُّّّّْْْْْْْْْْْْْْْْْْ\r\n\r\n' ; cat ) | nc 172.245.22.211 80`

Test #2 : 
`echo -n $'POST #123# HTTP/1.1\r\nTransfer-Encoding: chunked\r\n\x20\xd9\x8c\xd9\x8f\xd9\x92\xd9\x92\xd9\x92\xd9\x92\xd9\x92\xd9\x92\xd9\x92\xd9\x92\xd9\x92\xd9\x92\xd9\x92\xd9\x92\xd9\x92\xd9\x91\xd9\x91\xd9\x91\xd9\x91\xd9\x92\xd9\x92\xd9\x92\xd9\x92\xd9\x8f\xd9\x8f\xd9\x8f\xd9\x92\xd9\x8f\xd9\x8f\r\n\r\n3\r\none\r\n3\r\ntwo\r\n5\r\nthree\r\n' | nc 172.245.22.211 80`

Test #3: 
`curl -s -X $'GET #123# HTTP/1.1\r\nX-Meta:' -T - -H 'Host:' -H 'User-Agent:' -H 'Accept:' -H 'Expect:' -H $'Trailer: 111\r\n\x20\xd9\x8c\xd9\x8f\xd9\x92\xd9\x92\xd9\x92\xd9\x92\xd9\x92\xd9\x92\xd9\x92\xd9\x92\xd9\x92\xd9\x92\xd9\x92\xd9\x92\xd9\x92\xd9\x91\xd9\x91\xd9\x91\xd9\x91\xd9\x92\xd9\x92\xd9\x92\xd9\x92\xd9\x8f\xd9\x8f\xd9\x8f\xd9\x92\xd9\x8f\xd9\x8f' 172.245.22.211`




As you can see this http request do not finish with `\r\n\r\n` , it finish with `\r\n ٌُُُُُُّّّّْْْْْْْْْْْْْْْْْْ\r\n\r\n` this has NON-ASCII characters after the first `\r\n`  +` space` , this NON-ASCII character is the trick to bypass the firewall.

The second confirmation is that someone create a android app that create a tunnel and bypass the firewall to connect to internet without restriction , this guy use a HTTP request on port 80 and shadowsocks on other port ( example 443 ) , but he do not tell us how he do it and the android app is clocked , that is the reason i want you to help me .

By the way the app that that guy create works on many countries . i capture the packet using TCPDUMP , this way you can see how it works on the screenshots or using Wireshark .

![4Full_request_and_response](https://user-images.githubusercontent.com/6626845/87883689-406fd580-c9c6-11ea-9417-dc0aacc2bc18.jpg)
![1HTTP REQUEST](https://user-images.githubusercontent.com/6626845/87883694-42399900-c9c6-11ea-9474-0e257b96925c.jpg)
![2NON-ASCII CHARACTERS](https://user-images.githubusercontent.com/6626845/87883697-449bf300-c9c6-11ea-9ba9-99fbfebe79cb.jpg)
![3HTTP RESPONSE](https://user-images.githubusercontent.com/6626845/87883702-4665b680-c9c6-11ea-896f-e29ac19352b7.jpg)

can you create a tunnel like this , ? i think you have the knowledge.

If you need more info , please let me know . I know you can do it.

**Comment by jpillora** [comment]
Looks like you need to hand-craft HTTP (chisel already issues a GET so this is not enough), so I'm sure its possible, though out of scope for chisel.
